### ISSUE TITLE
Add script to change dashboard used in the agent.

### ISSUE THREAD

**Issue by amoss** [netdata/netdata#8353]
##### Feature idea summary

We need to deploy the new react-based dashboard so that we can receive feedback from users. This is a substantial change to the code-base so we want an easy way to revert to the existing dashboard if any problems are discovered after it is deployed.

A new script will be added `netdata-switch-dashboard.sh` to `sbin`. 
* Accept a single parameter with two posible values: `react` and `classic`.
* Update the files in the target web-directory, i.e. overwrite the `index.html` and add/remove any additional files that are required, e.g. the webpack files for the react dashboard. The addition and removal will ensure that there is no accidental cross-use that will break after the old version is permanently retired.
* Because the behaviour will change depending on the installation prefix, it will need to be a `netdata-switch-dashboard.sh.in` file in the source repo that is setup with the correct paths at install time.
* The default install will be the new react dashboard, the script will allow a return to the classic dashboard.

##### Expected behavior

* After installation the dashboard can be selected by running the script.
* No restart of the agent should be necessary (it may be necessary to flush the local browser cache to see the changes).
* Testing feedback will be provided by *everybody* on the core-team within 24 hours of the PR being available.
* We will review the results of the testing in the following standup. Any blocking bugs will be fixed before the PR is merged.

**Comment by amoss** [comment]
@cakrit / @cosmix / @jacekkolasa / @Ferroin Please check this description and explicitly approve it with a comment below before the work starts.

**Comment by cakrit** [comment]
I only disagree with the use of sbin

**Comment by thiagoftsm** [comment]
A complement, bugs should be report on `netdata/dashboard` repository.
I agree with Chris, considering that `www` has the same owner of netdata, we do not need sbin.

**Comment by amoss** [comment]
The `sbin` directory is assumed to be in the path, I don't think that plugins.d has the same assumptions. I don't mind where it is installed as long as the decision is taken before the the implementation starts :).

**Comment by Ferroin** [comment]
Note that if we're removing the files from the dashboard that isn't in use, this will require significantly more work than just replicating what we had in #8212  I'm not saying I disagree with the reasoning on that aspect, just that it will take me longer to get it working since it will probably involve changing where the old dashboard actually gets installed in the resulting system.

Is there any interest other than my own for having a switch on the installer to just skip the dashboard install? There are a couple of use-cases for Netdata that don't really need the dashboard and the handling of the new dashboard makes it much easier to actually implement the option to just not install the dashboard at all.

**Comment by amoss** [comment]
My mental of how this would work is that we have three locations:
1. The target webroot in the installation prefix.
2. The source location for the old dashboard.
3. The source location for the new dashboard.

I'm viewing it as basically `rm -rf`ing the current target webroot and `cp -r` one of the two sources based on the argument. Is this a naive view of how this will work - is there some complexity that I'm missing when I think about it?

**Comment by Ferroin** [comment]
@amoss That's kind of what I was thinking too.

Does the web server properly handle the case of the web root being a symbolic link? If so, that would actually make this exponentially easier to do, switching dashboards would just be updating the symlink.

**Comment by amoss** [comment]
That raises an interesting problem - what happens if the user deletes the source directory after install but decides to switch the dashboard?

Perhaps we should put both versions in the target prefix instead.

**Comment by Ferroin** [comment]
Both side-by-side is what the earlier PR had. It works reasonably well, provided you use the right index.

**Comment by jacekkolasa** [comment]
that will work, thanks @Ferroin !

**Comment by cosmix** [comment]
I disagree with the coexistence of both dashboards in the 'live' (web root) folder. I'm good with copying the files from the 'selected' source folder to the web root.

If the user decides to delete one of the source directories (either) that's their problem and our script should fail. I wouldn't put both versions in the web root; it's cleaner if we have a clean directory with the 'live' version and might preempt issues with the coexistence of both versions.

I'm good with everything else stated here.

**Comment by jacekkolasa** [comment]
I just want to point out one more time that it's crucial to have the script work in both ways. This way it will be easy for us & users to test and compare both versions.

**Comment by Ferroin** [comment]
Based on apparent consensus, I'm starting on this now using the original proposal outlined here with `sbin` changed to our libexec directory.

I'll be reusing some of the code that as done for the original PR to save some time, but still expect that this will probably take me at least the rest of the day to get working properly with appropriate testing.

I'll open a draft PR attached to this issue shortly so that progress can be tracked more concretely.

**Comment by Ferroin** [comment]
Also, note that the initial implementation will _not_ cover handling this for binary packages. There's enough complexity there that we _really_ need to cover them with a separate issue (I'll open one for that once development of this is finished).

**Comment by Ferroin** [comment]
PR is ready for review, I'm adding everyone on the core team as reviewers, as well as @jacekkolasa and the appropriate code owners.
